File: TripPlanningService/Models/Destination.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TripPlanningService.Models
{
    public class Destination
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();

        [Required]
        public Guid TripPlanId { get; set; }

        [ForeignKey(nameof(TripPlanId))]
        public TripPlan TripPlan { get; set; }

        [Required]
        [MaxLength(150)]
        public string Name { get; set; }

        [MaxLength(200)]
        public string Location { get; set; }

        public DateTime ArrivalDate { get; set; }
        public DateTime DepartureDate { get; set; }

        public string Description { get; set; }
    }
}

File: TravelPlannerFrontend/src/models/Destination.js
export function createDestination(data = {}) {
  return {
    id: data.id ?? null,
    tripPlanId: data.tripPlanId ?? null,
    name: data.name ?? '',
    location: data.location ?? '',
    arrivalDate: data.arrivalDate ?? '',
    departureDate: data.departureDate ?? '',
    description: data.description ?? '',
  };
}